### ISSUE TITLE
Variable list or Example template for CMIP7/ScenarioMIP harmonization

### ISSUE THREAD

**Issue by jkikstra** [IAMconsortium/concordia#63]
Hi @coroa (and @gidden),

I have started work with MESSAGE colleagues to create input data to work with for concordia .
I.e., I spoke to a colleague about the CDR variables, and this morning with other colleagues about reporting Aircraft emissions.
They want to know exactly what they should be working towards. And I expect more IAM teams to be asking similar questions. 

> So basically, I would like to discuss what exactly I should be asking from IAM teams to report, as a minimum input. I.e., an "expected template" that needs to be filled by everyone.

**Example data**
To start from, I used shared an April version of RESCUE data.
However, since then, there's been a CEDS update, and you have made some changes.
I see a few variable naming schemes have been updated ([commit/PR](https://github.com/IAMconsortium/concordia/pull/62)).
Anything I should be aware of, for what data to report?

**Full variable list**
Maybe this one is more for @gidden.
To me it is not yet entirely clear what the full list of emissions variables is that we are taking as inputs, and require from models, for CMIP7/ScenarioMIP. 
Combined CEDS + GFED.

Is it indeed the following:
```
CEDS+|9+ Sectors|Emissions|*|Aggregate - Agriculture and LUC
CEDS+|9+ Sectors|Emissions|*|Agricultural Waste Burning
CEDS+|9+ Sectors|Emissions|*|Agriculture
CEDS+|9+ Sectors|Emissions|*|Aircraft
CEDS+|9+ Sectors|Emissions|*|CDR Afforestation
CEDS+|9+ Sectors|Emissions|*|CDR BECCS
CEDS+|9+ Sectors|Emissions|*|CDR DACCS
CEDS+|9+ Sectors|Emissions|*|CDR EW
CEDS+|9+ Sectors|Emissions|*|CDR Industry
CEDS+|9+ Sectors|Emissions|*|CDR OAE Uptake Ocean
CEDS+|9+ Sectors|Emissions|*|Energy Sector
CEDS+|9+ Sectors|Emissions|*|Forest Burning
CEDS+|9+ Sectors|Emissions|*|Grassland Burning
CEDS+|9+ Sectors|Emissions|*|Industrial Sector
CEDS+|9+ Sectors|Emissions|*|International Shipping
CEDS+|9+ Sectors|Emissions|*|OAE Calcination Emissions
CEDS+|9+ Sectors|Emissions|*|Residential Commercial Other
CEDS+|9+ Sectors|Emissions|*|Solvents Production and Application
CEDS+|9+ Sectors|Emissions|*|Total
CEDS+|9+ Sectors|Emissions|*|Transportation Sector
CEDS+|9+ Sectors|Emissions|*|Waste
```

The REMIND data does not report all of this (some are zero - what is done then), and there was a note on the difference between "CO2|Deforestation and other LUC" for CO2 and "*|Aggregate - Agriculture and LUC" for other gases. And how does it relate to "Agricultural Waste Burning", "Agriculture", "Forest Burning", "Grassland Burning"? 

What should sum up to be the same?
Is there input data tests that we can learn from? Or that we should write?

I also remember seeing "Peat Burning" too in all the variabledefs CSV files. (e.g., "[variabledefs-cmip6.csv](https://github.com/IAMconsortium/concordia/blob/main/data/variabledefs-cmip6.csv)").

"[variabledefs-rescue.csv](https://github.com/IAMconsortium/concordia/blob/main/data/variabledefs-rescue.csv)" seems a bit different, because it has a separation between "Modelled" and "Non-Modelled" in "Energy Sector" - which does not seem to be reported in the REMIND data.

**Confirming where only global is taken as input, and where more**
Aircraft and International shipping are global.
The rest is regional.
Following the variabledefs files.

**Comment by coroa** [comment]
What is used for RESCUE is defined in [variabledefs-rescue.csv](https://github.com/IAMconsortium/concordia/blob/main/data/variabledefs-rescue.csv) (as you correctly surmised):

## `CO2`

| sector                              | resolution                        |
| ----------------------------------- | --------------------------------- |
| Deforestation and other LUC         | global                            |
| Agriculture                         | global (regional would also work) |
| Aircraft                            | global                            |
| Energy Sector\|Modelled             | regional                          |
| Energy Sector\|Non-Modelled         | regional                          |
| Industrial Sector                   | regional                          |
| International Shipping              | global                            |
| Residential Commercial Other        | regional                          |
| Solvents Production and Application | regional                          |
| Transportation Sector               | regional                          |
| Waste                               | regional                          |
| OAE Calcination Emissions           | regional                          |
| CDR Afforestation                   | regional                          |
| CDR BECCS                           | regional                          |
| CDR DACCS                           | regional                          |
| CDR EW                              | regional                          |
| CDR Industry                        | regional                          |
| CDR OAE Uptake Ocean                | regional                          |

Everything to be given in units of `Mt CO2/yr`.

## `BC`, `CH4`, `CO`, `NH3`, `NOx`, `OC`, `Sulfur`, `VOC`
 
| sector                              | resolution |
| ----------------------------------- | ---------- |
| Agricultural Waste Burning          | regional   |
| Agriculture                         | regional   |
| Aircraft                            | global     |
| Energy Sector\|Modelled             | regional   |
| Energy Sector\|Non-Modelled         | regional   |
| Forest Burning                      | regional   |
| Grassland Burning                   | regional   |
| Industrial Sector                   | regional   |
| International Shipping              | global     |
| Peat Burning                        | regional   |
| Residential Commercial Other        | regional   |
| Solvents Production and Application | regional   |
| Transportation Sector               | regional   |
| Waste                               | regional   |

Everything to be given in units of `Mt {species}/yr`.

What comprises these sectors can also be seen in the attached CEDS mapping. ![[ceds_mapping.xlsx]]
The split of the `Energy Sector` was due to REMIND not having any representation of fossil flaring (`7A_Fossil-fuel-fires`) which became `Energy Sector|Non-Modelled` (assumed to be 0 in REMIND). This part was then adjusted to the CEDS variable by harmonisation. After harmonisation it is aggregated up with the `Energy Sector|Modelled` variable to `Energy Sector`, from which it is downscaled. 

## `N2O`, `HFC`, `C2F6`, `CF4`, `SF6`

Are only treated as a global gas total (and expected in the units below)

| HFC  | Mt CO2eq/yr |
| ---- | ----------- |
| C2F6 | kt C2F6/yr  |
| CF4  | kt CF4/yr   |
| SF6  | kt SF6/yr   |
| N2O  | kt N2O/yr   |


# To-be changed for CMIP7 (IMHO)

- Use N2O from new CEDS with the same sector mapping as `CH4`, `BC` and the others.
- Remove split of Energy Sector
- Consider making CO2 Agriculture regional (it was global since it was part of `Aggregate - Agriculture and LUC` for which only global historical data from GCB was available).

**Comment by coroa** [comment]
If a variable is missing from the model file it is (after issuing a warning) added as constant 0.

The current `model_zero` harmonization (which triggers when the scenario data is constantly 0) is to keep the historical base-year value for the full time horizon.

**Comment by coroa** [comment]
We added recently a separate `RESCUE|OAE|Alkalinity Addition` variable to provide the total alkalinity addition to the ocean for Ocean Alkalinity enhancement-based CDR in the units of `Tmol / yr`, but I think that is very rescue specific.

**Comment by jkikstra** [comment]
Thanks a lot @coroa. 

Follow-up question: in [ceds_mapping.xlsx](https://github.com/user-attachments/files/16936637/ceds_mapping.xlsx), I see "Biomass Burning" (CEDS: 3F_Agricultural-residue-burning-on-fields).
But this is not mentioned in your list, and I also don't see it elsewhere in this repository. I also assume it would have overlap with the 4 GFED categories?

**Comment by coroa** [comment]
> Follow-up question: in [ceds_mapping.xlsx](https://github.com/user-attachments/files/16936637/ceds_mapping.xlsx), I see "Biomass Burning" (CEDS: 3F_Agricultural-residue-burning-on-fields).
> But this is not mentioned in your list, and I also don't see it elsewhere in this repository. I also assume it would have overlap with the 4 GFED categories?
>
> EDIT: P.S. Oliver Fricko says that "Biomass Burning" indeed should be the same as "Agricultural Waste Burning"

Yes, your answer matches my understanding. We use GFED for all the non-CO2 burning history and GCB for CO2 (Deforestation and LUC, as well as CDR Afforestation).
